regex-capi/src/rure.rs

```diff
@@ -570,3 +570,22 @@ ffi_fn! {
         unsafe { (*re).len() }
     }
 }
+
+ffi_fn! {
+    fn rure_escape(pattern: *const c_char, length: size_t) -> *const c_char {
+        let pat : &[u8] = unsafe{ slice::from_raw_parts(pattern as *const u8, length) };
+        let str_pat = str::from_utf8(pat).unwrap();
+        let esc_pat = regex::escape(str_pat);
+        let c_esc_pat = CString::new(esc_pat).unwrap();
```

Comment:
I don't think _either_ of these unwraps are acceptable here unfortunately. Both are errors that need to be handled because they can be triggered by user input.

Reply:
I think we should use `rure_error`.

Reply:
If you like, you can add an `rure_escape_must` that will abort the program on error, similar to `rure_compile_must`. If we do that, you might consider copying the compilation API and using `uint8_t *` for `rure_escape` and `char *` for `rure_escape_must`.